Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
Nevertheless, `curl -I -L` will print all redirections while we only catch one `Location:`. And it is contradicted to the recursive in the code.